### PR TITLE
Adjusting HTML and adding schema

### PR DESCRIPTION
**Pull request by jodiedoubleday**
#### What does this PR do?
As our react components will be used on SEO sites we need to ensure they are fully schema'd up for search engines.
This PR add's in schema and cleans up the html a little.

#### What unit or integration tests does this PR have?
N/A

#### What selenium tests does this PR have?
N/A

#### How should a developer review this?
Take a look at https://schema.org/Rating which describes how to add schema to a rating/review
- We are somewhat unusual in our approach as we use a percentage for customers who would book again, a bit of a weird way to review a product.

#### How should this be manually tested?
One you have checked out the branch
`npm install`
`npm run build-dev`
and open `dev/index.html` to view the component

#### Any background context you want to provide?
We are making components for _all the things_

#### What are the relevant tickets?
TBC

#### What gif best describes how you feel about this work?
the reviews

---
#### Quality checklist (for PR author to complete BEFORE code review):
- [x] I've checked there is appropriate unit test coverage.
- [x] I've checked there is appropriate selenium test coverage.
- [x] I've included regression tests for any bug fixes.
- [x] I've updated documentation with any changes to procedures.
- [x] I've tested this on mobile Android, Safari, Chrome for any visual changes.
- [x] I've tested this on desktop IE9/10/11, Firefox, Chrome for any visual changes.
- [x] I've checked any significant visual changes with a member of the UX/UI pod.
- [x] I've checked this work against the requirements of the Jira.
- [x] I've added and updated translations for all supported languages
- [x] I've informed those who need to know of any unforeseen impact of this work (database migrations, deployment procedure changes).
- [x] I am ready for this to be code reviewed, merged and tested.
- [x] I've thought about writing a blog post about this

#### IP or Developer Review:
- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve run all the tests locally and they pass.
- [x] +1 from me!

#### Software Engineer or Developer review:
- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve run all the tests locally and they pass.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] +1 from me!

#### Software Engineer or project guru review:
- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve run all the tests locally and they pass.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

### CODE REVIEW

**Comment by nickine** [comment]
I'll take developer review

**Comment by lukehansell-hx** [comment]
We could add an onClick handler to the "Reviews" text and if present display it as a link, otherwise display it as plain text. Though this could go in another PR. I'm not sure about where to put the issue though... on this repo or in JIRA? If in JIRA then I'm not sure whose responsibility that would be...

**Comment by lukehansell-hx** [comment]
I'll take SE/Guru

File: code/templates/ReviewsComponentTemplate.jsx [hunk 1]
```diff
@@ -2,15 +2,10 @@ var React = require('react');
...
-          <span>{ props.reviewPercentage }%</span> would book again<br/>
-          ({ props.reviewCount }
-          <a title="Customer reviews"> Reviews</a>
-          )
-        </p>
-      </div>
+    <div itemProp="aggregateRating" itemScope itemType="http://schema.org/AggregateRating" className="ui-component-reviews">
+      <meta itemProp="bestRating" content="100" />
+      <div><span itemProp="ratingValue">{ props.reviewPercentage }</span>% would book again</div>
+      <div><span itemprop="reviewCount">({ props.reviewCount }</span> <a title="Customer reviews"> Reviews</a>)</div>
```
Comment (lukehansell-hx) [inline code/templates/ReviewsComponentTemplate.jsx]: itemprop > itemProp

Comment (lukehansell-hx) [inline code/templates/ReviewsComponentTemplate.jsx]: static text like this makes it impossible for us to do translations, could this text be dynamic? Maybe passed as the child?

File: code/templates/ReviewsComponentTemplate.jsx [hunk 2]
```diff
@@ -2,15 +2,10 @@ var React = require('react');
...
-        <p>
-          <span>{ props.reviewPercentage }%</span> would book again<br/>
-          ({ props.reviewCount }
-          <a title="Customer reviews"> Reviews</a>
-          )
-        </p>
-      </div>
+    <div itemProp="aggregateRating" itemScope itemType="http://schema.org/AggregateRating" className="ui-component-reviews">
+      <meta itemProp="bestRating" content="100" />
+      <div><span itemProp="ratingValue">{ props.reviewPercentage }</span>% would book again</div>
```
Comment (lukehansell-hx) [inline code/templates/ReviewsComponentTemplate.jsx]: Same static text problem. We need this to be dynamic if w're going to do translations...

**Comment by jodiedoubleday** [comment]
Ive not changed any of the text that wan't already there, just changed HTML structure and schema. I think getting rid of static text would be a different PR.

I did have `itemprop` rather than `itemProp` but react threw a warning saying that wasn't valid :cry:

**Comment by lukehansell-hx** [comment]
:+1: but could you raise a JIRA or an issue on this repo for the translation stuff.

**Comment by nickine** [comment]
Schema testing flagged that "no reviewed item specified".  Standalone that would be invalid but as long as schema is applied to the parent templates ui-component-reviews will take on the scope of the item it's within so itemReviewed will fall back to name of the item type.

File: __tests__/ReviewsComponent-test.js [hunk 3]
```diff
@@ -13,8 +13,8 @@ describe('ReviewsComponent', function() {
       <ReviewsComponent reviewPercentage={98} reviewCount={123} />
     );
 
-    var reviewText = TestUtils.findRenderedDOMComponentWithTag(reviewView, 'p');
-    expect(reviewText.getDOMNode().textContent).toEqual('98% would book again(123 Reviews)');
+    var reviewText = TestUtils.findRenderedDOMComponentWithClass(reviewView, 'ui-component-reviews');
+    expect(reviewText.getDOMNode().textContent).toEqual('98% would book again(123  Reviews)');
```
Comment (nickine) [inline __tests__/ReviewsComponent-test.js]: Do you need a double space here?

File: code/templates/ReviewsComponentTemplate.jsx [hunk 4]
```diff
@@ -2,15 +2,10 @@ var React = require('react');
...
+      <div><span itemProp="reviewCount">({ props.reviewCount }</span> <a title="Customer reviews"> Reviews</a>)</div>
```
Comment (nickine) [inline code/templates/ReviewsComponentTemplate.jsx]: Additional space in here too.

**Comment by nickine** [comment]
Apart from the weird double spacing it's all cool with me

**Comment by andrewcashmore** [comment]
Taking IP/Dev

**Comment by jodiedoubleday** [comment]
@lukehansell-hx issues created - https://github.com/holidayextras/ui-component-reviews/issues/2

**Comment by andrewcashmore** [comment]
:+1:

**Pull request merged**